FreeRTOS/FreeRTOS-Plus-TCP
# [IPv6] Fix coverity

[IPv6] Fix coverity

Description
-----------

Fix coverity violations introduced recently.
- DEADCODE in FreeRTOS_Routing.c [line 824-838](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/933/files#diff-cabc7b626b59843c03a1b0a4018200289c01a46d67b68e3445249b18db94a853R824-R838).
- MISRA C-2012 Rule 10.4 in FreeRTOS_RA.c [line 270](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/933/files#diff-9e4e339dd72a2f9dff410f7ad25a2653c08c721e41a336bc026af38331c729b3R270).
- MISRA C-2012 Rule 10.8 in FreeRTOS_IP_Utils.c [line 433](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/933/files#diff-2ce8c151306b92beaed7018536d4bf8a7e7d8b4f0769f86f29740e70b0baf2a5R433).
- MISRA C-2012 Rule 11.3 in FreeRTOS_TCP_State_Handling_IPV4.c [line 88-90](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/933/files#diff-3e8b149dd8c82b122e2dc2cb8419c48b63b270a4ab0587fbdbf0f80b2b0d0679R88-R90).
- MISRA C-2012 Rule 11.9 in FreeRTOS_Routing.c [line 743](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/933/files#diff-cabc7b626b59843c03a1b0a4018200289c01a46d67b68e3445249b18db94a853R743).
- MISRA C-2012 Rule 14.2 in FreeRTOS_ND.c [line 237](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/933/files#diff-0b96f56bea14d6da5cdc4b896fa39770842f4f130bdd9327590b2f76b586157eR237).
- MISRA C-2012 Rule 17.2 in FreeRTOS_Sockets.c [line 4636-4639](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/933/files#diff-255cfb289125afea00b5a770dbc33f43e1c85e502c78f9300fbe5425e52e6158R4636-R4639).
- MISRA C-2012 Rule 21.6 in FreeRTOS_Routing.c [line 1522-1524](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/933/files#diff-cabc7b626b59843c03a1b0a4018200289c01a46d67b68e3445249b18db94a853R1522-R1524).
- MISRA C-2012 Rule 8.13 in:
  - FreeRTOS_TCP_Utils_IPV6.c [line 66](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/933/files#diff-1ade6a004fb4a0987691c6f9e2495603c4955373016bf036c16f946252fa17ebR66).
  - FreeRTOS_ARP.c [line 479](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/933/files#diff-9f198f8c6d0910ca3a809f5ea578599dd4a738c11cd0fcec9f24d3de463b59dcR479).
- MISRA C-2012 Rule 9.2 in FreeRTOS_DNS_Cache.c [line 54](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/933/files#diff-b1c2ecc6f93ad22bb803ccf33d38843ae713a8b58063f9ef2baa843d7386f904R54) + FreeRTOS_IP.c [line 530-535](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/933/files#diff-8b61e00e54b55092bc30ba18e0dc9570da3218e0b816bd15d47ba3b81c054318R530-R535).
- MISRA C-2012 Rule 9.3 in FreeRTOS_Routing.c [line 743-744](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/933/files#diff-cabc7b626b59843c03a1b0a4018200289c01a46d67b68e3445249b18db94a853R743-R744).
- MISRA C-2012 Rule 11.3 in FreeRTOS_ARP.c [line 1471](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/933/files#diff-9f198f8c6d0910ca3a809f5ea578599dd4a738c11cd0fcec9f24d3de463b59dcR1471).
- MISRA C-2012 Rule 15.7 inFreeRTOS_DHCPv6.c [line 354](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/933/files#diff-e217be213cd6a95f1b0ad333bae888b14894c98e620f319f3b0efefb289af4e0R354).

Test Steps
-----------

```
cov-configure --force --compiler cc --comptype gcc  -- -m32 -std=gnu90

cmake -B build -S test/Coverity

cd build/
cp ../test/Coverity/coverity_misra.config .

cov-build --emit-complementary-info --dir cov-out make

cd cov-out/
cov-analyze --dir . --coding-standard-config ../coverity_misra.config --tu-pattern "file('.*/FreeRTOS-Plus-TCP/source/.*')"

cov-format-errors --dir . --exclude-files '(.*/FreeRTOS-Kernel/.*)' --html-output html-output
```

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
